packages/nightingale-linegraph-track/src/nightingale-linegraph-track.js

```diff
@@ -0,0 +1,197 @@
+      .append("path")
+      .attr("class", "graph")
+      .attr("id", (d) => d.name)
+      .attr("d", (d) => {
+        d.colour = d.colour || interpolateRainbow(Math.random()); // eslint-disable-line no-param-reassign
+        return this.drawLine(d)(d.values);
+      })
+      .attr("fill", "none")
+      .attr("stroke", (d) => d.colour)
+      .attr("transform", "translate(0,0)");
+
+    const mouseG = this.chartGroup
+      .append("g")
+      .attr("class", "mouse-over-effects");
+
+    const lines = this.chartGroup.selectAll("path.graph");
+
+    const mousePerLine = mouseG
+      .selectAll(".mouse-per-line")
+      .data(this._data)
+      .enter()
+      .append("g")
+      .attr("class", "mouse-per-line");
+
+    mousePerLine
+      .append("circle")
+      .attr("r", 7)
+      .style("stroke", (d) => {
+        return d.colour;
+      })
+      .style("fill", "none")
+      .style("stroke-width", "1px")
+      .style("opacity", "0");
+
+    mousePerLine.append("text").attr("transform", "translate(10,3)");
+
+    const _this = this;
+    mouseG
+      .append("rect") // append a rect to catch mouse movements on canvas
+      .attr("width", this.width) // can't catch mouse events on a g element
+      .attr("height", this._height)
+      .attr("fill", "none")
+      .attr("pointer-events", "all")
+      .on("mouseout", () => {
+        // on mouse out hide circles and text
+        d3.selectAll(".mouse-per-line circle").style("opacity", "0");
+        d3.selectAll(".mouse-per-line text").style("opacity", "0");
+      })
+      // .on('mouseover', function() { // on mouse in show circles and text
+      //   d3.selectAll(".mouse-per-line circle")
+      //       .style("opacity", "1");
+      //   d3.selectAll(".mouse-per-line text")
+      //       .style("opacity", "1");
+      // })
+      .on("mousemove", () => {
+        // mouse moving over canvas
+        const mouse = d3.mouse(this);
+
+        if (
+          mouse[0] < _this.xScale(_this.beginning) ||
+          mouse[0] > _this.xScale(_this.end) + _this.getSingleBaseWidth()
+        ) {
+          d3.selectAll(".mouse-per-line circle").style("opacity", "0");
+          d3.selectAll(".mouse-per-line text").style("opacity", "0");
+        } else {
+          d3.selectAll(".mouse-per-line circle").style("opacity", "1");
+          d3.selectAll(".mouse-per-line text").style("opacity", "1");
+
+          const features = [];
+          const seqPosition = Math.floor(_this.xScale.invert(mouse[0]));
+
+          d3.selectAll(".mouse-per-line text").text((d) => {
+            const value = d.values.find((v) => v.position === seqPosition);
+            features.push(value);
+            return value ? value.value : "";
+          });
+
+          d3.selectAll(".mouse-per-line").attr("transform", (d, i) => {
+            let beginning = 0;
+            let end = lines.nodes()[i].getTotalLength();
+            let target = null;
+            let pos = {};
+            while (true) {
```

Comment:
Could you add some comments about what this is doing (trying to find the closest point to the mouse pointer in the paths by using iterative dichotomy) and the link to the d3 example?

Reply:
Hi Xavier, yes, it is doing the same. I will add the comment and the link too.